### ISSUE TITLE
Issue with kubewarden-controller ARM image

### ISSUE THREAD

**Issue by tmuntaner** [kubewarden/kubewarden-controller#39]
Whenever I try to run the ARM image on an ARM server, I get a user process error.

The ARM image in question is from https://github.com/orgs/kubewarden/packages/container/kubewarden-controller/2933672?tag=v0.2.3.

## AWS T4G server

```
$ uname -a
Linux ip-172-31-1-59 5.3.18-24.64-default #1 SMP Tue May 4 10:05:02 UTC 2021 (0a8fae2) aarch64 aarch64 aarch64 GNU/Linux
```

```
$ docker run --rm ghcr.io/kubewarden/kubewarden-controller:v0.2.3@sha256:7d13230789b10bc9cc96b050c28bced40eb9d3541c286088c2c45199d1b67b8f
standard_init_linux.go:211: exec user process caused "exec format error"
```

## My x86_64 laptop with same ARM image

```
$ uname -a
Linux linux-h8ph 5.12.4-1-default #1 SMP Tue May 18 15:37:31 UTC 2021 (cf6ea2f) x86_64 x86_64 x86_64 GNU/Linux
```
```
$ docker run --rm ghcr.io/kubewarden/kubewarden-controller:v0.2.3@sha256:7d13230789b10bc9cc96b050c28bced40eb9d3541c286088c2c45199d1b67b8f
Unable to find image 'ghcr.io/kubewarden/kubewarden-controller:v0.2.3@sha256:7d13230789b10bc9cc96b050c28bced40eb9d3541c286088c2c45199d1b67b8f' locally
ghcr.io/kubewarden/kubewarden-controller@sha256:7d13230789b10bc9cc96b050c28bced40eb9d3541c286088c2c45199d1b67b8f: Pulling from kubewarden/kubewarden-controller
537c34dcb2f4: Pull complete 
bd3e6820895b: Pull complete 
Digest: sha256:7d13230789b10bc9cc96b050c28bced40eb9d3541c286088c2c45199d1b67b8f
Status: Downloaded newer image for ghcr.io/kubewarden/kubewarden-controller@sha256:7d13230789b10bc9cc96b050c28bced40eb9d3541c286088c2c45199d1b67b8f
WARNING: The requested image's platform (linux/arm64) does not match the detected host platform (linux/amd64) and no specific platform was requested
2021-06-10T12:28:50.713Z	ERROR	controller-runtime.client.config	unable to get kubeconfig	{"error": "invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable", "errorCauses": [{"error": "no configuration has been provided, try setting KUBERNETES_MASTER environment variable"}]}
github.com/go-logr/zapr.(*zapLogger).Error
	/go/pkg/mod/github.com/go-logr/zapr@v0.2.0/zapr.go:132
sigs.k8s.io/controller-runtime/pkg/client/config.GetConfigOrDie
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.7.2/pkg/client/config/config.go:154
main.main
	/workspace/main.go:83
runtime.main
	/usr/local/go/src/runtime/proc.go:204
```

With x86_64, docker complains that it's an ARM image, but the binary runs anyways without a user process error.

I think that this ARM image has a compiled x86_64 binary by mistake.

Thanks

**Comment by ereslibre** [comment]
Waiting for the build of the `v0.3.1` image. I will release a new chart after is published.

**Comment by ereslibre** [comment]
This is fixed now :tada: 

```
ereslibre@desktop ~> docker manifest inspect ghcr.io/kubewarden/kubewarden-controller:v0.3.1
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 739,
         "digest": "sha256:ad1b5172020682b599a1d3cf922c4482c29c07733692b5688512121697e8925b",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 739,
         "digest": "sha256:2c377fe68b808ae8d661a1ec674206a4fa3004dbf861e86123b14023737e1615",
         "platform": {
            "architecture": "arm64",
            "os": "linux"
         }
      }
   ]
}

ereslibre@desktop ~> docker create ghcr.io/kubewarden/kubewarden-controller@sha256:2c377fe68b808ae8d661a1ec674206a4fa3004dbf861e86123b14023737e1615 controller
Unable to find image 'ghcr.io/kubewarden/kubewarden-controller@sha256:2c377fe68b808ae8d661a1ec674206a4fa3004dbf861e86123b14023737e1615' locally
ghcr.io/kubewarden/kubewarden-controller@sha256:2c377fe68b808ae8d661a1ec674206a4fa3004dbf861e86123b14023737e1615: Pulling from kubewarden/kubewarden-controller
537c34dcb2f4: Already exists 
743e64b09af6: Pull complete 
Digest: sha256:2c377fe68b808ae8d661a1ec674206a4fa3004dbf861e86123b14023737e1615
Status: Downloaded newer image for ghcr.io/kubewarden/kubewarden-controller@sha256:2c377fe68b808ae8d661a1ec674206a4fa3004dbf861e86123b14023737e1615
WARNING: The requested image's platform (linux/arm64) does not match the detected host platform (linux/amd64) and no specific platform was requested
3211c0b82dbf73cae0bb8f9068e0704672bab256314ebcb932ed7f1a06f6af2a
ereslibre@desktop ~> docker cp 3211c0b82dbf73cae0bb8f9068e0704672bab256314ebcb932ed7f1a06f6af2a:/manager .
ereslibre@desktop ~> file manager
manager: ELF 64-bit LSB executable, ARM aarch64, version 1 (SYSV), statically linked, not stripped
```

Released the `v0.2.1` helm chart that points to this controller version :clap:
